plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -801,13 +827,13 @@ void producer_plugin::plugin_initialize(const boost::program_options::variables_
    my->_incoming_block_subscription = app().get_channel<incoming::channels::block>().subscribe([this](const signed_block_ptr& block){
       try {
          my->on_incoming_block(block);
-      } FC_LOG_AND_DROP();
+      } LOG_AND_DROP();
```

Comment:
This is still swallowing `std::bad_alloc` by re-throwing it up to :

https://github.com/EOSIO/appbase/blob/4b95cd33659e18bd4f6ea2327202cfa20b09c658/include/appbase/channel.hpp#L28-L30

What is the expected resolution when a `std::bad_alloc` is thrown?  should we call a method like `chain_plugin::handle_db_exhaustion()` specific to heap exhaustion?  is this only thrown from heap exhaustion?

Reply:
more detail, appbase's channels are not meant to handle exceptions thrown out of subscriptions by default.   We can install handlers on these channels that do something specific with exceptions (other than swallowing them) but it is likely better practice to sanitize the handlers such that they don't throw.